modules/eso-external-secret/main.tf

```
@@ -17,9 +17,9 @@ locals {
   # dockerjsonconfig secrets chain flag
   is_dockerjsonconfig_chain = length(var.es_container_registry_secrets_chain) > 0 ? true : false
 
-  # validation for dockerjsonconfig secrets chain -> if it is a chain the kube secret type must be dockerconfigjson and sm secret type iam_credentials
-  validate_condition_chain = local.is_dockerjsonconfig_chain == true && (var.es_kubernetes_secret_type != "dockerconfigjson" || var.sm_secret_type != "iam_credentials") # checkov:skip=CKV_SECRET_6: does not require high entropy string as is static value
-  validate_msg_chain       = "If the externalsecret is expected to generate a dockerjsonconfig secrets chain the only supported value for es_kubernetes_secret_type is dockerconfigjson and for sm_secret_type is iam_credentials"
+  # validation for dockerjsonconfig secrets chain -> if it is a chain the kube secret type must be dockerconfigjson and sm secret types iam_credentials, trusted_profile
+  validate_condition_chain = local.is_dockerjsonconfig_chain == true && (var.es_kubernetes_secret_type != "dockerconfigjson" || (var.sm_secret_type != "iam_credentials" && var.sm_secret_type != "trusted_profile")) # checkov:skip=CKV_SECRET_6: does not require high entropy string as is static value
```

what's the purpose of this check? I would expect it to be an or `(var.sm_secret_type != "iam_credentials" || var.sm_secret_type != "trusted_profile")` according to the comment

---

> what's the purpose of this check? 

This check only allows either iam_credentials or trusted_profile for dockerconfigjson chain list

> I would expect it to be an or (var.sm_secret_type != "iam_credentials" || var.sm_secret_type != "trusted_profile") according to the comment

  ```
validate_condition_chain = local.is_dockerjsonconfig_chain == true && (var.es_kubernetes_secret_type != "dockerconfigjson" || (var.sm_secret_type != "iam_credentials" && var.sm_secret_type != "trusted_profile"))
```

If this block of code turns out of to be true, then it should throw an error saying that dockerconfigjson is only valid for the types iam_credentials and trusted_profile. 

So going further into the code block, this code block should be returning true when the secret type is not iam_credentials and trusted profile as they are only supported for dockerconfigjson chain. So I have added the and block